### PR TITLE
Create checksum validate, log download and order info

### PR DESCRIPTION
**Pull request by GabrielDeveloper**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Issues         | https://mundipagg.atlassian.net/browse/MOD-340, https://mundipagg.atlassian.net/browse/MOD-350, https://mundipagg.atlassian.net/browse/MOD-342
| What?         | Create maintenance feature
| Why?          | to check the validity of module, to have access to the logs and see the order informations
| How?          | Created some endpoints to return the informations needed

### CODE REVIEW

File: integrityDeploy.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+//setting parameters based on modman file.
+$modmanFilePath = 'modman';
+$modmanRawData = file_get_contents($modmanFilePath);
+$rawLines = explode("\n",$modmanRawData);
```
Comment (ianlessa) [inline integrityDeploy.php]: faltando espaço

File: integrityDeploy.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+foreach ($rawLines as $rawLine) {
+    if (
+        substr($rawLine,0,1) !== '#' &&
+        strlen($rawLine) > 0
+    ) {
+        $line = array_values(array_filter(explode(' ',$rawLine)));
```
Comment (ianlessa) [inline integrityDeploy.php]: faltando espaço

File: integrityDeploy.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        if ($line[0] === 'modman') {
+            $integrityEngineBasePath = explode('/',$line[1]);
```
Comment (ianlessa) [inline integrityDeploy.php]: faltando espaço

File: integrityDeploy.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            array_pop($integrityEngineBasePath);
+            $integrityEngineBasePath = './' . implode('/',$integrityEngineBasePath) . '/';
```
Comment (ianlessa) [inline integrityDeploy.php]: faltando espaço

File: integrityDeploy.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+   die("error on setting parameters!");
+}
+
+//creating moduleFiles md5;
+$autoload = $argv[1];
+require_once $autoload;
+$integrityEngine = new \Mundipagg\Integrity\IntegrityEngine();
+
+$modmanRawData = file_get_contents($modmanFilePath);
+$rawLines = explode("\n",$modmanRawData);
```
Comment (ianlessa) [inline integrityDeploy.php]: já sabe, né?

File: integrityDeploy.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+$directoriesIgnored = explode(',', $argv[2]);
+
+$listFiles = $integrityEngine->getFileListFromArrayData($rawLines, $directoriesIgnored);
+$integrityData = $integrityEngine->generateMD5FromArray($listFiles);
+
+file_put_contents($integrityFilePath,json_encode($integrityData));
```
Comment (ianlessa) [inline integrityDeploy.php]: espaço maroto

File: system/library/mundipagg/src/Integrity/README.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#Maitenance (XNine)
```
Comment (ianlessa) [inline system/library/mundipagg/src/Integrity/README.md]: Documentações do XNine devem ir para wiki do plug team

File: system/library/mundipagg/src/Integrity/SystemInfoInterface.php [hunk 8]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Mundipagg\Integrity;
+
+interface SystemInfoInterface
+{
+
```
Comment (ianlessa) [inline system/library/mundipagg/src/Integrity/SystemInfoInterface.php]: linha extra

File: system/library/mundipagg/src/Helper/OpencartOrderInfo.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
+<?php
+
```
Comment (michelpl) [inline system/library/mundipagg/src/Helper/OpencartOrderInfo.php]: O Ian pediu pra colocar um comentário aqui

**Pull request merged**